cmd/k-license/k-license_test.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"testing"
+)
+
+var opt_test = &Options{}
+
+// Running E2E Test
+func TestAddLicense(t *testing.T) {
+	opt_test.templatesDir = "../../hack/boilerplate"
+	opt_test.excludeDirs = excludeDirsLocations
+	opt_test.path = "../"
+	opt_test.confirm = false
+	opt_test.Run()
```

Comment:
2 things:
- Can we confirm that no changes have been made? 
  - We can have a `./k-license/testdata` dir and have a simple Go file there for now without a licence header.
  - One test case: set `confirm`, verify changes happen
  - One test case: set `confirm`, verify no changes happen
  - [Not needed in this PR, but maybe a followup] the `testdata` can have some excluded files too and we can test for that.
- We need to check for error returned by `Run()` and fail the test if there is an error.